macosx/BadgeView.mm

```diff
@@ -24,6 +34,35 @@ - (instancetype)init
     {
         _fDownloadRate = 0.0;
         _fUploadRate = 0.0;
+
+        NSShadow* stringShadow = [[NSShadow alloc] init];
+        stringShadow.shadowOffset = NSMakeSize(2.0, -2.0);
+        stringShadow.shadowBlurRadius = 4.0;
+
+        _fAttributes = [[NSMutableDictionary alloc] initWithCapacity:3];
+        _fAttributes[NSForegroundColorAttributeName] = NSColor.whiteColor;
+        _fAttributes[NSShadowAttributeName] = stringShadow;
+
+        // make sure text fits on the badge
+        // DownloadBadge and UploadBadge should have the same size
+        NSSize badgeSize = [NSImage imageNamed:@"DownloadBadge"].size;
+        // text is centered, so we calculate size with arbitrary symmetric arrows
+        NSString* maxString = [NSString stringWithFormat:@" ▼ %@ ▼ ", [NSString stringForSpeedAbbrev:888.8]]; // "888.8 K" localized
+        CGFloat fontSize = 26.0;
+        NSSize stringSize;
+        do
+        {
+            _fAttributes[NSFontAttributeName] = [NSFont boldSystemFontOfSize:fontSize];
+            stringSize = [maxString sizeWithAttributes:_fAttributes];
+            fontSize -= 1.0;
+        } while (badgeSize.width < stringSize.width);
+
+        // DownArrowTemplate and UpArrowTemplate should have the same size
+        NSSize arrowImageSize = kWhiteDownArrow.size;
+        NSSize spaceCharacterSize = [@" " sizeWithAttributes:_fAttributes];
```

Comment:
Isn't it a bit hacky way to do this? :)

Reply:
I've been giving Antoine lessons, are they paying off?

Reply:
- I'm unsure if the badge is the same size in all display resolutions
- I'm unsure if the string is using the same system font for everybody, including people using accessibility features

So I tried to set the position of the arrow according to the font attributes.
But well, we could alternatively hardcode an inset proportional to the badge height, ignoring non standard system fonts.

Reply:
OK, I went through all accessibility settings, and none of them affects the default system font, so I guess that I don't need to worry about that, and I can just do an inset based on the badge size.

Reply:
Let me know if 065ba8e57cdf438fcec7e38807f9e05335a03277 is less hacky.